### PR TITLE
Development

### PR DESCRIPTION
**Pull request by John-Arboleda**
# Ruby Enumerable Methods

This pull request is an exercise of the Ruby language basics. The main goal of the project is to replicate the following methods:
`#each`, `#each_with_index`, `#select`, `#all?`, `#any?`, `#none?`, `#count`, `#map`, `#inject`. The method identical to `#each` would be `#my_each`, for `#each_with_index` would be the method `#my_each_with_index`, and so on and so forth.

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  # my_each method
```
Comment (cyonii) [inline enumerable.rb]: Kindly remove these comments and others like them as they are not important.

**Review by cyonii — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️✍️

Hello @John-Arboleda & @errea, you have done really a great job. Out of 79 test cases run against your code, you only have 11 failures, this is a lot, you should give yourselves pats in the back 🤓.

## To Improve

- [x] Your README appears to be a copy of the original project's requirements, please make your own descriptive README and document your work in it.
- [x] The names of your methods are descriptive in themselves, therefore there's is no point writing the same method names as comments, please remove those as they make your code messy.

> **Failed tests**

Out of your `11` failed tests, `5` was as a result of the question mark sign `?` in your `#my_count` methods. Please change `#my_count?` to `#my_count`. Review the code below and make sure your code passes these tests.

```ruby
# Note that these variables may be reused accross the examples
truthy_array = [1, true, 'hi', []]
falsy_array = [nil, false, nil, false]
words = %w[dog door rod blade]
```

### **`#my_all?`**

- [x] when no block or argument is given, it returns true when none of the collection members are false or nil

```ruby
# Expectation
truthy_array.my_all? == truthy_array.all?
```

- [x] when a Regex is passed as an argument, it returns true if all of the collection matches the Regex

```ruby
# Expectation
words.my_all?(/d/) == words.all?(/d/)
```

### **`#my_any?`**

- [x] when no block or argument is given, it returns false if at least one of the collection is not true

```ruby
# Expectation
falsy_array.my_any? == falsy_array.any?
```

- [x] when a Regex is passed as an argument, it returns true if any of the collection matches the Regex

```ruby

# Expectation
words.my_any?(/d/) == words.any?(/d/)
```

### **`#my_none?`**

- [x] when no block or argument is given, it returns true only if none of the collection members is true

```ruby
# Expectation
falsy_array.my_none? == true
```

- [x] when a Regex is passed as an argument returns false if any of the collection matches the Regex

```ruby
words.my_none?(/d/) == words.none?(/d/)
```

I hope these test cases helps.
Happy coding 🤓💻

### You can find me on slack 👉 [CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) if you need further assistance or clarification. I'm always available and will be glad to help.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @John-Arboleda and @errea,  Both of you have done well on this project, just 3 tests cases are failing, your progress is indeed commendable. 👏 

Let's make a few changes, and we will be done.

- [x] Going forward your commit messages should be capitalized, and written in the imperative tense. You can watch this [video](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6) on how to edit your commit messages using git rebase
 See [Git Commit Best Practices](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices#formatting-rules) for more insights

- [x] `#my_all?` when a Regex is passed as an argument it should return true if all of the collection matches the Regex
```ruby
%w[dog door rod blade].my_all?(/d/)  # => true
```

- [x] `#my_any?` when a Regex is passed as an argument returns true if any of the collection matches the Regex
```ruby
%w[dog door rod blade].my_any?(/d/)  # => true
```

- [x] `#my_none?` when a Regex is passed as an argument returns false if any of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/d/)  # => false
```

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by mke2111 — APPROVED** [review]
# STATUS: APPROVED ㊗️ 🟡 

## Hey Happy Coder, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval.

### You may now merge to the main branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

## Reach me on slack for a chit-chat 🆘  Roy Mukuye

**Pull request merged**